scheduler/src/managers/scheduler/schedule.rs

```diff
@@ -0,0 +1,40 @@
+use crate::managers::scheduler::errors::SchedulerError;
+use orka_proto::node_agent::{Workload, WorkloadStatus as AgentWorkloadStatus};
+use orka_proto::scheduler_controller::workload_status::Resources;
+use orka_proto::scheduler_controller::{
+    SchedulingRequest, WorkloadStatus as SchedulerWorkloadStatus,
+};
+
+pub fn map_request_to_workload(
+    scheduling_request: SchedulingRequest,
+) -> Result<Workload, SchedulerError> {
+    let workload = scheduling_request.workload.map(|w| Workload {
+        name: w.name,
+        image: w.image,
+        environment: w.environment,
+        r#type: w.r#type,
+        resource: None, // TO CHANGE
+    });
+    return workload.ok_or(SchedulerError::InvalidWorkload("".to_string()));
+}
+
+pub fn map_agent_workload_status_to_controller_workload_status(
+    agent_workload: AgentWorkloadStatus,
+) -> SchedulerWorkloadStatus {
+    // Cast Agent resources usage to Controller resources usage
+    let agent_resource_usage = agent_workload.resource.map(|r| Resources {
+        cpu: r.usage.clone().unwrap_or_default().cpu,
+        memory: r.usage.clone().unwrap_or_default().memory,
+        disk: r.usage.unwrap_or_default().disk,
+    });
+    // Cast Agent Workload status to Controller workload status
+    let controller_workload = {
+        SchedulerWorkloadStatus {
+            name: agent_workload.name,
+            status_code: agent_workload.status,
+            resource_usage: agent_resource_usage,
+            message: agent_workload.message,
+        }
+    };
+    return controller_workload;
```

Comment:
In this function, as well as others, you should use the "last expression return" ([Rust lang doc](https://doc.rust-lang.org/std/keyword.return.html)) to produce clearer code.
I also suggest an improvement in the way you handle agent resource mapping.

```suggestion
    SchedulerWorkloadStatus {
        name: agent_workload.name,
        status_code: agent_workload.status,
        message: agent_workload.message,
        resource_usage: agent_workload.resource.map(|r| {
            let usage = r.usage;
            Resources {
                cpu: usage.unwrap_or_default().cpu,
                memory: usage.unwrap_or_default().memory,
                disk: usage.unwrap_or_default().disk,
            }
        }),
    }
```

Reply:
Done